src/content/1.7/modules/sample_modules/order_view_page_new_hooks/view-order-hooks-signature-widget.md

```diff
@@ -0,0 +1,448 @@
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoViewOrderHooks\Repository;
+
+use DateTimeImmutable;
+use Db;
+use Order as PrestaShopOrder;
+use PrestaShop\Module\DemoViewOrderHooks\Collection\Orders;
+use PrestaShop\Module\DemoViewOrderHooks\DTO\Order;
+
+class OrderRepository
+{
+    /**
+     * @var Db
+     */
+    private $db;
+
+    public function __construct()
+    {
+        $this->db = Db::getInstance();
+    }
+
+    /**
+     * Get all orders that a customer has placed.
+     */
+    public function getCustomerOrders(int $customerId, array $excludeOrderIds = []): Orders
+    {
+        $orders = PrestaShopOrder::getCustomerOrders($customerId);
+        $ordersCollection = new Orders();
+
+        foreach ($orders as $order) {
+            if (in_array($order['id_order'], $excludeOrderIds)) {
+                continue;
+            }
+
+            $ordersCollection->add(new Order(
+                (int) $order['id_order'],
+                $order['reference'],
+                (int) $order['current_state'],
+                new DateTimeImmutable($order['date_add'])
+            ));
+        }
+
+        return $ordersCollection;
+    }
+}
+
+```
+Let's create `OrderSignaturePresenter` class responsible for returning order customer data  
+in `src/Presenter/`:
+
+```php
+<?php
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoViewOrderHooks\Presenter;
+
+use Gender;
+use Order;
+use PrestaShop\Module\DemoViewOrderHooks\Entity\OrderSignature;
+
+class OrderSignaturePresenter
+{
+    /**
+     * @var string
+     */
+    private $signatureImgDir;
+
+    public function __construct(string $signatureImgDir)
+    {
+        $this->signatureImgDir = $signatureImgDir;
+    }
+
+    public function present(OrderSignature $signature, int $languageId): array
+    {
+        $order = new Order($signature->getOrderId());
+        $customer = $order->getCustomer();
+        $gender = new Gender($customer->id_gender, $languageId);
+
+        return [
+            'firstName' => $customer->firstname,
+            'lastName' => $customer->lastname,
+            'gender' => $gender->name,
+            'filename' => $this->signatureImgDir.$signature->getFilename()
+        ];
+    }
+}
+```
+Then lets use Symfony Dependency Injection 
+(https://www.freecodecamp.org/news/a-quick-intro-to-dependency-injection-what-it-is-and-when-to-use-it-7578c84fa88f/).
+and create services configuration for the above 
+classes in `config/services.yml`. For more information: 
+https://devdocs.prestashop.com/1.7/modules/concepts/services/#symfony-services
+The intent behind dependency injection is to achieve Separation of Concerns of construction and use of objects. 
+This can increase readability and code reuse, reduce dependencies, lead to more testable code.
+```yaml
+parameters:
+  signatureImgDirectory: 'signatures/'
+
+services:
+  prestashop.module.demovieworderhooks:
+    class: DemoViewOrderHooks
+    factory: [Module, getInstanceByName]
```

Comment:
This one is very smart as well!!

Although I took a look at the `Module::getInstanceByName` which is a bit frightening 😅

It uses the very old legacy custom DependencyInjection of PrestaShop, and if the modules is instanciated elsewhere the module instances collection is not updated, so we could have two instances of the same module. This will need to be improved in the core.